cmd/maddy-pam-helper/main.go

```diff
@@ -0,0 +1,20 @@
+package main
+
+/*
+#cgo LDFLAGS: -lpam
+#cgo CFLAGS: -DCGO
```

Comment:
Can we build with `-Wall -Wextra -Wpedantic -std=c11`?

Reply:
It will emit a lot of warnings for cgo wrapper.

Reply:
And it turns out we need `-std=gnu18` just because I used getline.

Reply:
>It will emit a lot of warnings for cgo wrapper.

Oh and add `-Werror` to the mix too

>And it turns out we need -std=gnu18 just because I used getline. Should I replace it?

No no, `getline` is POSIX. You just need to `#define _POSIX_C_SOURCE 200809L` at the top of the file before any `#include`.

(See the manpage)

Reply:
>It will emit a lot of warnings for cgo wrapper.

It's not very sane to write a security-oriented program in C without these

Reply:
We probably don't even need `-std=c11`, `-std=c99` will probably be enough

Reply:
Will add `-Wno-error=unused-parameter` to make warnings from cgo wrappers non-fatal.

Reply:
Oh, this one is very annoying. Yeah just add `-Wno-unused-parameter` so we don't even see warnings